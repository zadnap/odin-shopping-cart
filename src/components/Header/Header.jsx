import styles from './Header.module.scss';
import Input from '@/components/Input/Input';
import Button from '@/components/Button/Button';
import { Link, useNavigate } from 'react-router-dom';
import logo from '@/assets/logo.png';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faBars, faCartShopping } from '@fortawesome/free-solid-svg-icons';
import { useState } from 'react';

function Header({ isOpenNav, setIsOpenNav }) {
  const [query, setQuery] = useState('');
  const navigate = useNavigate();

  const handleSearch = (e) => {
    e.preventDefault();

    if (query.trim()) {
      navigate(`/search?query=${encodeURIComponent(query)}`);
    }
  };

  return (
    <header className={styles.header} role="banner">
      <Link className={styles.logo} to="/">
        <img className={styles.logoImage} src={logo} alt="Homepage" />
        <span className={styles.title}>Cinemart</span>
      </Link>
      <form role="search" className={styles.searchForm} onSubmit={handleSearch}>
        <label htmlFor="header-search" className="sr-only">
          Search Movies
        </label>
        <Input
          placeholder="Movies, series, shows..."
          id="header-search"
          value={query}
          onChange={(e) => setQuery(e.target.value)}
        />
      </form>
      <div className={styles.buttonGroup}>
        <Link
          className={styles.cartLink}
          to="/cart"
          aria-label="View Shopping Cart"
        >
          <FontAwesomeIcon icon={faCartShopping} />
        </Link>
        <Button
          className={styles.menu}
          aria-label="Open Navigation Menu"
          aria-expanded={isOpenNav}
          aria-controls="primary-navigation"
          onClick={() => setIsOpenNav((prev) => !prev)}
        >
          <FontAwesomeIcon icon={faBars} />
        </Button>
      </div>
    </header>
  );
}

export default Header;
