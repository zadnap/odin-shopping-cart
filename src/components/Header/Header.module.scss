.header {
  display: flex;
  justify-content: space-between;
  padding: 10px 20px;
  position: sticky;
  top: 0;

  .searchBar {
    width: 40%;
  }
}
