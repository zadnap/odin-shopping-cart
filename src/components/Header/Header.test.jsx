import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Header from './Header';

describe('Header component', () => {
  it('should render to match snapshot', () => {
    const { container } = render(
      <MemoryRouter>
        <Header />
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
  });
});
