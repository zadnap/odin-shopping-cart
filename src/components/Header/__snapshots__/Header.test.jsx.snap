// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header component > should render to match snapshot 1`] = `
<div>
  <header
    class="_header_d975ee"
  >
    <a
      class="_logo_5ff345"
      data-discover="true"
      href="/"
    >
      <img
        alt="Homepage"
        class="_logoImage_5ff345"
        src="/src/assets/logo.png"
      />
      <h1
        class="_title_5ff345"
      >
        Cinemart
      </h1>
    </a>
    <div
      class="_inputContainer_9af53b"
    >
      <span
        data-testid="fa-icon"
      />
      <input
        class="_input_9af53b"
        placeholder="Movies, series, shows..."
        type="text"
      />
    </div>
    <button
      class="_button_ee73f7  "
    >
      <span
        data-testid="fa-icon"
      />
    </button>
  </header>
</div>
`;
